### ISSUE TITLE
SNOW-855574: Creating a package: TypeError: 'SnowflakeCursor' object is not subscriptable

### ISSUE THREAD

**Issue by n-batalha** [Snowflake-Labs/snowcli#224]
### Python version

3.8.13

### Platform

macOS-13.4.1-x86_64-i386-64bit

### What happened

Installed the CLI:

```
pipx install "snowflake-cli-labs" --python python3.8
```

I was able to use it before, but after the upgrade to 1.0.0, when running:

```
snow --debug snowpark package upload --environment dev -f $wheel -s example_stage --overwrite
```

It fails, see full error below. The error is specifically on `deploy_response[6]`. This `deploy_response` is a cursor. It shouldn't be subscriptable?

Relevant LOC:

* from here https://github.com/Snowflake-Labs/snowcli/blob/v1.0.0/src/snowcli/cli/snowpark/package.py#L141
* https://github.com/Snowflake-Labs/snowcli/blob/v1.0.0/src/snowcli/snow_connector.py#L160-L173
* https://github.com/Snowflake-Labs/snowcli/blob/v1.0.0/src/snowcli/snow_connector.py#L938

Similar error here: https://github.com/Snowflake-Labs/snowcli/issues/39

### Console output

```Shell
─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ /Users/nbatalha/.local/pipx/venvs/snowflake-cli-labs/lib/python3.8/site-packages/snowcli/cli/sno │
│ wpark/package.py:145 in package_upload                                                           │
│                                                                                                  │
│   142 │   │   print(type(deploy_response))                                                       │
│   143 │   │   print(dir(deploy_response))                                                        │
│   144 │   │   print(deploy_response)                                                             │
│ ❱ 145 │   │   log.info(f"Package {file} {deploy_response[6]} to Snowflake @{stage}/{file}.")     │
│   146 │   │   if deploy_response[6] == "SKIPPED":                                                │
│   147 │   │   │   log.info(                                                                      │
│   148 │   │   │   │   "Package already exists on stage. Consider using --overwrite to overwrit   │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
TypeError: 'SnowflakeCursor' object is not subscriptable
```

### How to reproduce

In macOS with Python 3.8 and pipx, run:

```
pipx install "snowflake-cli-labs" --python python3.8
# configure a connection named `dev` and a local package
snow --debug snowpark package upload --environment dev -f some_wheel -s public --overwrite
```

**Comment by n-batalha** [comment]
Testing previous versions, the issue occurs only in v1.0.0 (currently the latest). 0.2.9 is fine

**Comment by sfc-gh-jsikorski** [comment]
There`s a fix for the issue on main - will be included in next release

**Comment by sfc-gh-afoster** [comment]
Hi All - I reproduced this error, `Exception message: 'SnowflakeCursor' object is not subscriptable` and then confirmed the package was successfully uploaded, in spite of the error. It wasn't called out here if the stage was checked for the package that the user was attempting to upload, so just wanted to call this out.

Environment: Ubuntu 22.04
SnowCLI Version: 1.0.0
Python 3.10.6

Console output:
`╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ /home/parallels/myenv/lib/python3.10/site-packages/snowcli/cli/snowpark/package.py:141 in        │
│ package_upload                                                                                   │
│                                                                                                  │
│   138 │   │   │   │   role=conn.ctx.role,                                                        │
│   139 │   │   │   │   warehouse=conn.ctx.warehouse,                                              │
│   140 │   │   │   )                                                                              │
│ ❱ 141 │   │   log.info(f"Package {file} {deploy_response[6]} to Snowflake @{stage}/{file}.")     │
│   142 │   │   if deploy_response[6] == "SKIPPED":                                                │
│   143 │   │   │   log.info(                                                                      │
│   144 │   │   │   │   "Package already exists on stage. Consider using --overwrite to overwrit   │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
TypeError: 'SnowflakeCursor' object is not subscriptable
`

**Comment by sfc-gh-turbaszek** [comment]
Fixed in v1.0.1 via #227
